include/solvers/time_solver.h

```diff
@@ -279,7 +279,8 @@ class TimeSolver : public ReferenceCountedObject<TimeSolver>
   virtual Real last_complete_deltat();
 
   /**
-   * Setter and getter for the final time variable
+   * The final_time variable can be set by the user or in another library
+   * function. For use, see documentation of final_time variable.
```

Comment:
This tells us *what* we can do with it, but we didn't need the comment for that.  set_final_time() clearly sets a final_time() variable, get_final_time() clears one.  All is solid there.

What we're missing in the comment is *why* we want or need to do this.  Suppose I'm time marching to steady state (via some ||du/dt|| < tolerance).  I don't know my final time.  Is it impossible for me to do adjoint time stepping and QoI postprocessing?  What happens if I get final_time wrong?

Reply:
In the constructor, final_time is set to std::nan("1"). Not setting it does not preclude adjoint time stepping or QoI postprocessing. If you want to evaluate a final time QoI and compute a corresponding adjoint, you need to set this variable to a finite value.